src/main/scala/beam/physsim/jdeqsim/EventToHourFrequency.scala

```diff
@@ -51,10 +52,9 @@ class EventToHourFrequency(val controlerIO: OutputDirectoryHierarchy)
   override def notifyIterationEnds(event: IterationEndsEvent): Unit = {
     val filePath = controlerIO.getIterationFilename(event.getIteration, "PhysSimEventToHourFrequency.csv")
     val csvWriter = new CsvWriter(filePath, Vector("event_type", "hour", "count"))
-    val maxHour = eventToHourFreq.values.flatten.map(_._1).max
-    logger.info(s"Handled ${cnt} events. MaxHour: $maxHour")
-
     try {
+      val maxHour = SequenceUtils.maxOption(eventToHourFreq.values.flatten.map(_._1)).getOrElse(-1)
```

Comment:
Same as elsewhere, should this throw instead of resulting in an empty loop?

Reply:
if there are no maxHour then csv file won't be written. on the other hand if there is behavior does not change. In the previous code an exception is thrown. Considering this code is there for a long time I believe is working fine.
This rule is the same as replacing `last` with`drop(1)`.

Reply:
I think we should avoid throwing exception. is the same as replacing `tail` with `drop(1)`

Reply:
That is a larger debate - maybe worth bringing up at the next standup - I'm of the mindset that an exception early or at least a log for something that might result in unexpected output..but becomes hard to track or worse - isn't noticed at all.